### ISSUE TITLE
Discussion Episode 6, Part1: Continuous Integration (CI) & Continuous Delivery (CD) with Fernando Cejas (& EVERYTHING & LIFE)

### ISSUE THREAD

**Issue by artem-zinnatullin** [artem-zinnatullin/TheContext-Podcast#49]
Yo there, great episode, bad sound (@artem-zinnatullin bad as always lol), enjoy!

**Comment by android10** [comment]
It was good guys! Really enjoyed it. I need your permission to upload the file to SoundCloud in my profile. I was thinking also to ask for an account for you guys, so it makes easier to listeners to have access to the audio content. 
I think having everything here is very geed and cool but I would try to facilitate more its access. 

My two cents :). 

And again, thanks for the opportunity.

**Comment by sockeqwe** [comment]
Thanks for joining us. Really appreciated. Regarding uploading to your sound cloud profile: I'm fine with that! I think Artem is a little bit busy the next days, but I guess he will be fine with that too. 
Last word has Artem of course ...

**Comment by artem-zinnatullin** [comment]
I'm fine with uploading too, but I think will be better to give us the
access so we could manage it without need in pinging you each time!

**Comment by android10** [comment]
@artem-zinnatullin of course. That is what I meant. I'm trying to get you an account which will belong to you :). Let's see what happens. Will keep you guys posted.

**Comment by artem-zinnatullin** [comment]
Ah, okay. Don't worry if it won't be possible, thank you for doing this!

**Comment by sebaslogen** [comment]
Great discussion guys, I really enjoyed the episode:
- the team dynamics part: we also try to create this safe and positive environment in my team. One thing that is working very well lately is to add likes and genuine compliments to clever or creative pieces of code in code reviews, so not just comments for fixing stuff but also positive things that are worth mentioning
- feature toggles: I would love to use these but I have a lot of questions on how you actually build and use them. When you build a new feature that partially or completely modifies an existing feature, do you always cleanly refactor and isolate the two versions of the code? do you sometimes build a completely new module and then delete the old one?

I'm also positevely surprised by the audio quality of the episode, it's far for ideal but I could understand everything without problems (listened to half episode with headphones).

**Comment by artem-zinnatullin** [comment]
Hey hey @sebaslogen, thank you for feedback 💕!

>One thing that is working very well lately is to add likes and genuine compliments to clever or creative pieces of code in code reviews, so not just comments for fixing stuff but also positive things that are worth mentioning

Haha, I'd like to show you our comments in the PRs, they're full of meme pictures and trolling :trollface:  which works better than just compliments in my opinion, but for sure positive feedback in codereview is VERY important, especially in Open Source projects, it motivates people to do more and do it better!

>feature toggles: I would love to use these but I have a lot of questions on how you actually build and use them.

Oh yeah, that's important topic, looks like @android10 is most experienced in that between us (or maybe @sockeqwe??), guys please share your thoughts on that!

Regarding the sound quality, we're trying to make it better (@sockeqwe let's try to record sound separately on each end of the call?) , glad you found it good enough to listen to. Again thank you for feedback man!

**Comment by artem-zinnatullin** [comment]
@sebaslogen forgot to say that a @android10 was talking a little bit more about feature flags in second part of the episode (which I'm editing right now and hope we'll release it in couple days), so stay tuned!

**Comment by android10** [comment]
@sebaslogen 

> feature toggles: I would love to use these but I have a lot of questions on how you actually build and use them. When you build a new feature that partially or completely modifies an existing feature, do you always cleanly refactor and isolate the two versions of the code? do you sometimes build a completely new module and then delete the old one?

If a big refactor is required, I would do it in a separate PR as a preparation of the terrain. Otherwise as good boy scouts we included with the feature. It is worth mentioning that feature flags are usually at UI level and if you have your code correctly structured, you could for example inject different presenters depending on the state of the feature flag. 
Once the feature is up and running we clean up the dead code.

**Comment by sebaslogen** [comment]
@android10 thanks for the info. I really wish we were doing this in my current project because we haven't released in a few months for a big product/feature release and now I'm very scared a lot of tricky bugs will explode at once (plus no staged rollout allowed for this one).

@artem-zinnatullin now I'm very curious to see one of those PRs going off-rails with memes.
